airflow/hooks/jdbc_hook.py

```diff
@@ -25,64 +27,54 @@ class JdbcHook(BaseHook):
-        self.jdbc_driver_loc = jdbc_driver_loc
-        self.jdbc_driver_name = jdbc_driver_name
-
-        if (conn_id is None):
-            self.host = host
-            self.login = login
-            self.psw = psw
-            self.db = db
-            self.port = port
-            self.extra = extra
-        else:
-            conn = self.get_connection(conn_id)
-            self.host = conn.host
-            self.login = conn.login
-            self.psw = conn.password
-            self.db = conn.schema
-            self.port = conn.port
-            self.extra = conn.extra
-
-
-        self.jdbc_url = jdbc_url.format(self.host, self.port, self.db, self.extra)
+            self, *args, **kwargs):
+
+        super(JdbcHook,self).__init__(*args,**kwargs)
+
+        #conn_id = getattr(self, self.conn_name_attr)
+        #if (conn_id is None):
+        #    self.host = host
+        #    self.login = login
+        #    self.psw = psw
+        #    #self.db = db
+        #    #self.port = port
+        #    self.extra = extra
+        #    self.jdbc_driver_loc = jdbc_driver_loc
+        #   self.jdbc_driver_name = jdbc_driver_name
+        #else:
+        conn = self.get_connection(getattr(self, self.conn_name_attr))
+        self.host = conn.host
+        self.login = conn.login
+        self.psw = conn.password
+        #self.db = conn.schema
+        #self.port = conn.port
+        self.extra = conn.extra
+        self.jdbc_driver_loc = conn.extra_dejson.get('jdbc_drv_path')
+        self.jdbc_driver_name = conn.extra_dejson.get('jdbc_drv_clsname')
+
+
+        #self.jdbc_url = jdbc_url.format(self.host, self.port, self.db, self.extra)
 
     def get_conn(self):
-        logging.info("Trying to connect using jdbc url: " + self.jdbc_url)
         conn = jaydebeapi.connect(self.jdbc_driver_name,
-                           [str(self.jdbc_url), str(self.login), str(self.psw)],
+                           [str(self.host), str(self.login), str(self.psw)],
                                   self.jdbc_driver_loc,)
         return conn
 
-    def get_records(self, sql, autocommit=False):
-        '''
-        Executes the sql and returns a set of records.
-        '''
-        conn = self.get_conn()
-        conn.jconn.autocommit = autocommit
-        cur = conn.cursor()
-        cur.execute(sql)
-        rows = [] if not cur._rs else cur.fetchall()
-        cur.close()
-        conn.close()
-        return rows
-
-    def get_pandas_df(self, sql):
-        '''
-        Executes the sql and returns a pandas dataframe
-        '''
-        import pandas.io.sql as psql
-        conn = self.get_conn()
-        df = psql.read_sql(sql, con=conn)
-        conn.close()
-        return df
-
-    def run(self, sql, autocommit=False):
+    def run(self, sql, autocommit=False, parameters=None):
+        """
+        Runs a command
+        """
         conn = self.get_conn()
-        conn.jconn.autocommit = autocommit
+        if self.supports_autocommit:
```

Comment:
Eventually I'll change `DbApiHook` to define a `set_autocommit(conn)` that different hooks will have to override.